application-licensing-licensor/application-licensing-licensor-api/src/main/java/com/xwiki/licensing/internal/DefaultLicensedExtensionManager.java

```diff
@@ -112,4 +114,25 @@
 
         return Collections.emptyMap();
     }
+
+    @Override
+    public Collection<ExtensionId> getVisibleLicensedExtensions()
+    {
+        Collection<ExtensionId> paidExtensions = getLicensedExtensions();
+
+        Set<ExtensionId> visiblePaidExtensions = new HashSet<ExtensionId>(paidExtensions);
+        for (ExtensionId extensionId : paidExtensions) {
+            InstalledExtension installedExtension = installedExtensionRepository.getInstalledExtension(extensionId);
+
+            Collection<ExtensionDependency> dependencies = installedExtension.getDependencies();
+            for (ExtensionId paidExtension : paidExtensions) {
+                if (dependencies.stream().filter(o -> o.getId().contentEquals(paidExtension.getId())).findFirst()
```

Comment:
```suggestion
                if (dependencies.stream().filter(dependency -> dependency.getId().equals(paidExtension.getId())).findFirst()
```

Regarding ``contentEquals``, I don't think it's needed when both sides are Strings.